thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/common/BaseThirdEyeApplication.java

```diff
@@ -33,17 +27,43 @@
   protected TaskManager anomalyTaskDAO;
   protected WebappConfigManager webappConfigDAO;
   protected MergedAnomalyResultManager anomalyMergedResultDAO;
+  String mode = "hibernate";
 
   public void initDAOs() {
     String persistenceConfig = System.getProperty("dw.rootDir") + "/persistence.yml";
     LOG.info("Loading persistence config from [{}]", persistenceConfig);
-    PersistenceUtil.init(new File(persistenceConfig));
-    anomalyFunctionDAO = PersistenceUtil.getInstance(AnomalyFunctionManagerImpl.class);
-    anomalyResultDAO = PersistenceUtil.getInstance(RawAnomalyResultManagerImpl.class);
-    emailConfigurationDAO = PersistenceUtil.getInstance(EmailConfigurationManagerImpl.class);
-    anomalyJobDAO = PersistenceUtil.getInstance(JobManagerImpl.class);
-    anomalyTaskDAO = PersistenceUtil.getInstance(TaskManagerImpl.class);
-    webappConfigDAO = PersistenceUtil.getInstance(WebappConfigManagerImpl.class);
-    anomalyMergedResultDAO = PersistenceUtil.getInstance(MergedAnomalyResultManagerImpl.class);
+    if ("jdbc".equalsIgnoreCase(mode)) {
```

Comment:
We can merge DaoProvider and PersistenceUtil to a DaoFactory and abstract the datalayer initialization logic there.